src/core/histogram.cc

```diff
@@ -96,6 +104,65 @@ histogram::print(FILE *fp) const {
                                  CLASSIC);  // Format CLASSIC/CSV supported.
 }
 
+seastar::metrics::histogram
+histogram::get_seastar_metrics_histogram() const {
+  // logarithmic histogram configuration. this will range from 10 microseconds
+  // through around 6000 seconds with 26 buckets doubling.
+  //
+  // TODO:
+  //   1) expose these settings through arguments
+  //   2) upstream log_base fix for sub-2.0 values. in hdr the log_base is a
+  //   double but is truncated (see the int64_t casts on log_base below which is
+  //   the same as in the hdr C library). this means that if we want buckets
+  //   with a log base of 1.5, the histogram becomes linear...
+  const size_t num_buckets = 26;
+  const int64_t first_value = 10;
+  const double log_base = 2.0;
+
+  seastar::metrics::histogram sshist;
+  sshist.buckets.resize(num_buckets);
+
+  sshist.sample_count = hist_->sample_count;
+  sshist.sample_sum = (double)hist_->sample_sum;
+
+  // stack allocated; no cleanup needed
+  struct hdr_iter iter;
+  struct hdr_iter_log *log = &iter.specifics.log;
+  hdr_iter_log_init(&iter, hist_->hist, first_value, log_base);
```

Comment:
ah! this does not need a delete because it does not allocate memory

```C
void hdr_iter_init(struct hdr_iter* iter, const struct hdr_histogram* h)
{
    iter->h = h;

    iter->counts_index = -1;
    iter->total_count = h->total_count;
    iter->count = 0;
    iter->cumulative_count = 0;
    iter->value = 0;
    iter->highest_equivalent_value = 0;
    iter->value_iterated_from = 0;
    iter->value_iterated_to = 0;

    iter->_next_fp = _all_values_iter_next;
}


```